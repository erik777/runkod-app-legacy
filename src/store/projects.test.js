import reducer, {
  fetchAct,
  fetchedAct,
  fetchErrorAct
} from './projects';

import {
  logOutAct
} from './user';

let state = undefined;

it('1- default', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- fetch', () => {
  const act = fetchAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- fetch error', () => {
  const act = fetchErrorAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('4- fetched', () => {
  const act = fetchedAct([{attrs: {_id: 'project1'}}, {attrs: {_id: 'project2'}}]);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('5- user logout', () => {
  const act = logOutAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
