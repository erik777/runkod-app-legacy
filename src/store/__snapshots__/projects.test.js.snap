// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "selected": null,
}
`;

exports[`2- fetch 1`] = `
Object {
  "list": Array [],
  "loading": true,
  "selected": null,
}
`;

exports[`3- fetch error 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "selected": null,
}
`;

exports[`4- fetched 1`] = `
Object {
  "list": Array [
    Object {
      "_id": "project1",
      "redirectTo": "",
    },
    Object {
      "_id": "project2",
      "redirectTo": "",
    },
  ],
  "loading": false,
  "selected": null,
}
`;

exports[`5- user logout 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "selected": null,
}
`;
